design/m1_push_to_file_design.md

```diff
@@ -0,0 +1,1247 @@
+| conjur.org/<br />retry-count-limit      | RETRY_COUNT_LIMIT   | Defaults to 5
+| conjur.org/<br />retry-interval-sec     | RETRY_INTERVAL_SEC  | Defaults to 1 (sec)              |
+| conjur.org/<br />debug-logging          | DEBUG               | Defaults to `false`              |
+| **conjur.org/<br />connect-config-dir** | CONJUR_AUTHN_URL<br />CONJUR_URL<br />CONJUR_ACCOUNT<br />CONJUR_SSL_CERTIFICATE | Directory containing Conjur connection information. Connection parameters are written to files with names `CONJUR_AUTHN_URL`, `CONJUR_URL`, `CONJUR_ACCOUNT`, and `CONJUR_SSL_CERTIFICATE`. The directory must match the `mountPath` used for the Conjur Connect ConfigMap as shown in the [example Kubernetes manifests](example-helm-partial-named-template-for-secrets-provider-init-container) below. The default value is `/conjur/connect`. |
+
+#### Annotations for Push-to-File Secrets Mappings
+
+##### Secrets Groups
+
+The configuration of the Secrets Provider push-to-file secrets mappings
+makes use of "secrets groups". A secrets group is a logical grouping of
+application secrets, typically belonging to a particular component of an
+application deployment (e.g. all secrets related to a backend database).
+
+As shown in the table below, all push-to-file annotations contain a reference
+to a secret group, using the form: `conjur.org/<parameter-name>.<group>`.
+
+There is a one-to-one correspondence between secrets groups and files that
+are written to the application Pod's Conjur secrets shared volume.
+
+##### Supported Annotations
+
+| Annotation                                     | Description |
+|------------------------------------------------|-------------|
+| conjur.org/conjur-secrets.{secret-group}       | List of secrets to be retrieved from Conjur. Each entry can be either:<ul><li>A Conjur variable path</li><li> A key/value pairs of the form `<alias>:<Conjur variable path>` where the `alias` represents the name of the secret to be written to the secrets file |
+| conjur.org/conjur-secrets-path.{secret-group}  | Prefix to use for all Conjur variable paths for a secrets group. An example is `prod/backend/`. The path is assumed to be relative to root policy path. A leading `/` is ignored. A trailing `/` is optional. Defaults to the root policy path. |
+| conjur.org/secret-file-path.{secret-group}     | File path (including file name) for secrets file to be written. This is a required value if the `conjur.org/secret-file-template.{secret-group} annotation is configured for the secret group. The default file name is `{secret-group}.{secret-file-type}`, e.g. `database.yaml`, and the default directory is `/conjur/secrets/`. The directory must match the `mountPath` that is defined for the shared Volume used by the Secrets Provider to write secrets files (refer to [example VolumeMounts below](example-helm-partial-named-template-for-secrets-provider-init-container)) |
```

Comment:
do we want to support providing a path only here (no filename) and the default filename is used? 

can we add an error case for `secret-file-path` supplied but conflicts with `mountpath`? maybe this is already covered